packages/eslint-plugin/src/rules/no-unused-vars.ts

```diff
@@ -361,9 +398,18 @@ export default util.createRule<Options, MessageIds>({
         function getAssignedMessageData(
           unusedVar: TSESLint.Scope.Variable,
         ): Record<string, unknown> {
-          const additional = options.varsIgnorePattern
-            ? `. Allowed unused vars must match ${options.varsIgnorePattern.toString()}`
-            : '';
+          const def = unusedVar.defs[0];
+          let additional = '';
+
+          if (
+            options.destructuredArrayIgnorePattern &&
+            def &&
+            def.name.parent?.type === AST_NODE_TYPES.ArrayPattern
```

Comment:
Nit: could be simplified a teeny bit.

```suggestion
            def?.name.parent?.type === AST_NODE_TYPES.ArrayPattern
```